Made shortcuts to get in position to grab specimen and deliver specimen

arjavprabhu opened the pull request:
plz review coach tim and other people (WIP)

arjavprabhu committed:
Made shortcuts to get in position to grab specimen and deliver specimen

botnerd commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/control/ArmController.java:
```diff
@@ -114,4 +114,14 @@ public void ZeroLSEncoders() {
         linear_slide_motor.setMode(DcMotor.RunMode.STOP_AND_RESET_ENCODER);
         linear_slide_motor.setMode(DcMotor.RunMode.RUN_WITHOUT_ENCODER);
     }
+
+    public void GrabSpecimenPos() {
+        desiredLinearSlidePos = 700;
+        desiredShoulderPos = 350;
+    }
+
+    public void DeliverSpecimenPos() {
+        desiredLinearSlidePos = 100;
+        desiredShoulderPos = 1200;
+    }
```
We have to be careful about setting position directly since they could be far away from the current pos. This is because our PID settings could result in oscillations.  We can try it, and if it doesn't cause problems, we can use it. Otherwise, we will need to write some methods that increment to the positions we want.

arjavprabhu replied:
I wrote code on my computer that increments it, but I have commented it out as we have to test it tomorrow.

arjavprabhu committed:
Made shortcuts to get in position to grab specimen and deliver specimen

botnerd committed:
Add arm modes for setting positions for grabbing and delivering specimens

Also, add override button for allowing shoulder and linear slide to go past their min limits (for resetting zero).

botnerd committed:
Merge branch 'refs/heads/main' into team/specimen-shortcuts

# Conflicts:
#	TeamCode/src/main/java/org/firstinspires/ftc/teamcode/control/ArmController.java

botnerd committed:
Fix bug in arm mode (maybe)

botnerd committed:
Fix zero encoders (maybe)

botnerd reviewed (approved):
Good to go
